src/esn.calendar.libs/app/services/shells/calendar-shell.js

```diff
@@ -801,6 +802,48 @@ require('./valarm-shell.js');
       });
     }
 
+    /**
+     * Returns the differences between this and the object to compare
+     * @param  {CalendarShell} that an event
+     *
+     * @return {Array} an array containing the keys of differences between events.
+     */
+    function getChanges(that) {
```

Comment:
I haven't read the logic thoroughly, but at first glance this is quite similar to the `equals` method. So, can we reuse the logic?

Reply:
i think `equals` has other additional fields like `recurrenceId`.